lib/screens/util.ex

```diff
@@ -162,4 +164,34 @@ defmodule Screens.Util do
   def route_type_from_id("CR-" <> _), do: :commuter_rail
   def route_type_from_id("Boat-" <> _), do: :ferry
   def route_type_from_id(_), do: :bus
+
+  def outdated?(screen_id, client_refresh_timestamp) do
```

Comment:
To save on requests, what do you think about having the screens config cache GenServer incorporate the `LAST_DEPLOY` value into config? That is, it would fetch the config JSON as usual, but also fetch `LAST_DEPLOY`. Then, it would iterate through each screen in the config and set its `refresh_if_loaded_before` to the `LAST_DEPLOY` time if that's later than the stored timestamp.

This way, we would only need to check `LAST_DEPLOY` every 15 seconds when the cached config gets updated, rather than checking it once per screen data request.

Reply:
That's a great idea. I added a new struct attribute to `Screens.Config.State` that holds the timestamp and moved all of the `DateTime` comparison logic to `:refresh_if_loaded_before` in the GenServer.